web/src/controller/coaching_session_controller.rs

```diff
@@ -0,0 +1,53 @@
+use crate::controller::ApiResponse;
+use crate::extractors::{
+    authenticated_user::AuthenticatedUser, compare_api_version::CompareApiVersion,
+};
+use crate::{AppState, Error};
+use axum::extract::{Query, State};
+use axum::http::StatusCode;
+use axum::response::IntoResponse;
+use axum::Json;
+use entity_api::coaching_session as CoachingSessionApi;
+use service::config::ApiVersion;
+use std::collections::HashMap;
+
+use log::*;
+
+#[utoipa::path(
+    get,
+    path = "/coaching_sessions",
+    params(
+        ApiVersion,
+        ("coaching_relationship_id" = Option<Id>, Query, description = "Filter by coaching_relationship_id"),
+        ("from_date" = Option<NaiveDate>, Query, description = "Filter by from_date"),
+        ("to_date" = Option<NaiveDate>, Query, description = "Filter by to_date")
+    ),
+    responses(
+        (status = 200, description = "Successfully retrieved all Coaching Sessions", body = [entity::coaching_sessions::Model]),
+        (status = 401, description = "Unauthorized"),
+        (status = 405, description = "Method not allowed")
+    ),
+    security(
+        ("cookie_auth" = [])
+    )
+)]
+pub async fn index(
+    CompareApiVersion(_v): CompareApiVersion,
+    AuthenticatedUser(_user): AuthenticatedUser,
+    // TODO: create a new Extractor to authorize the user to access
```

Comment:
Is this comment still relevant?

Reply:
I have been copying these over from the other controllers as I believe it is relevant since we currently have no concept of authorization. Made sense to me to propagate the comment just so we would have a sense of where we might want authorization. Open to your thoughts there.

Reply:
Ok, so it does still apply. Very good.